### ISSUE TITLE
[test-suite] Automate file rename, rename again then publish (page->rename->rename->publish)

### ISSUE THREAD

**Issue by alhambrav** [craftercms/craftercms#2055]
Please automate page rename, rename again then publish (page>rename->rename->publish) **ID 43** in Studio -> Content.

**Setup**

1) Create a site using Editorial BP
2) Open the Sidebar if it's not yet open
3) Expand "Pages"
4) Right click on "articles/2016/12" and select "New Content"
5) Select "Page - Article" and click on the "Ok" button in the "Choose Content Type" screen
6) Put "foo" in the "Page URL" field
7) Put "foo" in the "Internal Name" and "Title" field
8) For "Categories", select "Entertainment"
9) Upload an image in the "Image" field
10) Fill in the rest of the required fields in the form and save your changes
11) From the Sidebar, navigate to "articles/2016/12/foo"
12) Right click on "foo" and select "Approve & Publish"
13) Select "Items should go live now" in the "Selected Item Scheduling" field then click on "Submit"
14) Setup your site for delivery by running the init-site script

Here's how your dashboard "My Recent Activity" widget should look like at this point:
<img width="867" alt="screen shot 2018-04-23 at 1 47 40 pm" src="https://user-images.githubusercontent.com/25483966/39143806-ee8cf3d6-46fc-11e8-9bcf-caeb754abe5e.png">

**Steps**

1) Login as admin user and open the site created in the setup
2) Open the Sidebar if it's not yet open
3) In Pages, navigate to "articles/2016/12/foo"
4) Right click on "foo" and select "Edit"
5) In the "Page Properties" section, click on the "Edit" button next to the "Page URL" field
6) Click "Ok" on the Warning message
7) Change the "Page URL" from "foo" to "bar"
8) Click on the "Save and Close" 
9) From the Sidebar, click on the page "foo" to preview
10) Click on the Dashboard
11) From the Sidebar, right click on the page "articles/2016/12/foo" and select "Edit"
12) In the "Page Properties" section, click on the "Edit" button next to the "Page URL" field
13) Click "Ok" on the Warning message
14) Change the "Page URL" from "bar" to "baz"
15) Click on the "Save and Close" 
16) From the Sidebar, click on the page "foo" to preview
17) Click on the Dashboard 
18) From the Sidebar, right click on the page "articles/2016/12/foo" and select "Approve & Publish"
19) Select "Items should go live now" then click on "Submit"
20) Wait for a minute or so to make sure publishing is done, click on the dashboard
21) In your browser, go to "localhost:9080?crafterSite=siteName", where `siteName` is the name given on the site we're working on and then click on "Entertainment" from the left-rail
22) Click on the article "foo"

**Expected Output**

4) Form for editing page "foo" should open
5) A warning message should appear after clicking on "Edit"
6) The "Page URL" field should now be editable and not grayed out
8) Page should return to previous view (Preview or Dashboard)
9) The URL should contain "page=/articles/2016/12/bar"
10) The "My Recent Activity" widget should contain the following:
```
    Item Name: foo      Edit: Edit    URL: /articles/2016/12/bar
    Item Name: foo      Edit: Edit    URL: /articles/2016/12/foo  
```
<img width="869" alt="screen shot 2018-04-23 at 1 49 24 pm" src="https://user-images.githubusercontent.com/25483966/39143909-32e5b7d4-46fd-11e8-8b62-7fd3bffe8b8b.png">

11) Form for editing page "foo" should open
12) A warning message should appear after clicking on "Edit"
13) The "Page URL" field should now be editable and not grayed out
15) Page should return to previous view (Preview or Dashboard)
16) The URL should contain "page=/articles/2016/12/baz"
17) The "My Recent Activity" widget should contain the following:
```
    Item Name: bar      Edit: Edit    URL: /articles/2016/12/bar
    Item Name: foo      Edit: Edit    URL: /articles/2016/12/baz
    Item Name: foo      Edit: Edit    URL: /articles/2016/12/foo  
```
<img width="860" alt="screen shot 2018-04-23 at 1 50 38 pm" src="https://user-images.githubusercontent.com/25483966/39143961-5a3867be-46fd-11e8-8e4b-51b815bd33af.png">

18) The "Approve for Publish" dialog should open
19) The "Approve for Publish" dialog should close
20) In your dashboard, the "Recently Published" widget should contain the following:
```
    Item Name: foo    Edit: Edit    URL: /articles/2016/12/baz
```
<img width="871" alt="screen shot 2018-04-23 at 1 51 47 pm" src="https://user-images.githubusercontent.com/25483966/39144006-86af4ae2-46fd-11e8-97f9-03ce6de0791a.png">

22) The page just published should be viewable on the delivery side, with the url http://localhost:9080/articles/2016/12/baz

**Comment by luishernandezmonge** [comment]
Start working on this.

**Comment by luishernandezmonge** [comment]
@alhambrav Done, please validate.

**Comment by alhambrav** [comment]
I am getting errors for the test case when I run it.  Here's the trace: https://gist.github.com/alhambrav/66a779192b96da52d001bb395b35c859

**Comment by luishernandezmonge** [comment]
Start working on the fix for this.

**Comment by luishernandezmonge** [comment]
@alhambrav Fixed on PR: craftercms/test-suite#341

**Comment by alhambrav** [comment]
Verified.
